packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -102,17 +136,37 @@ export class ModularChangeFamily
 	compose(changes: TaggedChange<ModularChangeset>[]): ModularChangeset {
 		let maxId = changes.reduce((max, change) => Math.max(change.change.maxId ?? -1, max), -1);
 		const genId: IdAllocator = () => brand(++maxId);
+		const crossFieldTable = newCrossFieldTable<ComposeData>();
 
 		const composedFields = this.composeFieldMaps(
 			changes.map((change) => tagChange(change.change.changes, change.revision)),
 			genId,
+			crossFieldTable,
 		);
+
+		while (crossFieldTable.fieldsToUpdate.size > 0) {
```

Comment:
Consider de-duplicating this fixed point iteration pattern could. No worries if it's too awkward.